src/lib/comms/include/sol-coap.h

```diff
@@ -480,29 +480,11 @@ int sol_coap_header_set_id(struct sol_coap_packet *pkt, uint16_t id);
  * @a addr. If the server cannot be created, NULL will be returned and
  * errno will be set to indicate the reason.
  *
- * @param addr The address where the server will listen on.
+ * @param transport The transport that will be used.
  *
  * @return A new server instance, or NULL in case of failure.
- *
- * @see sol_coap_secure_server_new()
- */
-struct sol_coap_server *sol_coap_server_new(const struct sol_network_link_addr *addr);
-
-/**
- * @brief Creates a new secure CoAP server instance.
- *
- * Creates a new, unsecured, CoAP server instance listening on address
- * @a addr. This server will encrypt communication with its endpoints
- * using DTLS. If the server cannot be created, NULL will be returned
- * and errno will be set to indicate the reason.
- *
- * @param addr The address where the server will listen on.
- *
- * @return A new server instance, or NULL in case of failure.
- *
- * @see sol_coap_server_new()
  */
-struct sol_coap_server *sol_coap_secure_server_new(const struct sol_network_link_addr *addr);
+struct sol_coap_server *sol_coap_server_new(struct sol_coap_transport *transport);
```

Comment:
document this function will take ownership of the transport, then there is no need to delete it. Even in the case of failure, so the below construct can be used:

```c
server = sol_coap_server(sol_coap_transport_new_udp(addr, false));
SOL_NULL_CHECK(server, -EINVAL);
```

of course static-inline helper `sol_coap_server_new_udp(addr, use_dtls)` to do the pair would be nice :-)